Bio/PDB/rotamers.py

```diff
@@ -0,0 +1,274 @@
+"""
+A tool that introduces a point mutation to a given residue based on the Dunbrack rotamer library.
+The tool uses sample residues stored in small PDB files, and uses a reduced rotamer library
+to save space (only rotamers with higher than 5% probability are present).
+The tool makes in place modification on a PDB structure object.
+"""
+
+from Bio.Entrez.Parser import ValidationError
+from Bio.PDB import Polypeptide
+from Bio.PDB.NeighborSearch import NeighborSearch
+from Bio.PDB.Selection import unfold_entities
+from Bio.PDB.vectors import calc_dihedral, Vector, rotaxis2m
+from Bio import SVDSuperimposer
+from collections import defaultdict
+from Bio.PDB.ic_data import (
+    CHI_ANGLES,
+    RESIDUE_ORDER,
+    VW_RADII,
+    residue_atom_bond_state,
+    ic_data_backbone,
+)
+from Bio.SeqUtils import seq3
+import numpy as np
+import os
+import gzip
+
+DATA_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)), "rotamers")
+DUNBRACK_FILTERED = f"{DATA_DIR}/dunbrack_reduced.gz"
+
+
+class Rotamer:
+    """
+    Object to store rotamers
+    """
+
+    def __init__(self, resname, psi, phi, chi1, chi2, chi3, chi4, probability):
+        """
+        Init the rotamer with given diherdral angles
+        """
+        self.psi = float(psi)
+        self.phi = float(phi)
+        self.chi1 = float(chi1)
+        self.chi2 = float(chi2)
+        self.chi3 = float(chi3)
+        self.chi4 = float(chi4)
+        self.resname = resname
+        self.probability = float(probability)
+        self.sample_residue = False
```

Comment:
OK, it looks mostly good now! If we’re keeping it as a standard class though, you might also want to change the `self.sample_residue = None` to represent the not-yet-set values, and then change `if self._sample_residue` with `if self._sample_residue is not None`. This is partly convention, but also it avoids any issues with objects that act like they are False, like empty dictionaries and similar

Reply:
I changed it to None and changed the logic as well. Also as per a previous comment I changed the name to `RotamerSampler` as it indeed sounds nicer.